apps/src/aichat/views/ChatWorkspace.tsx

```diff
@@ -1,35 +1,96 @@
-import React from 'react';
+import React, {useState} from 'react';
 import PanelContainer from '@cdo/apps/lab2/views/components/PanelContainer';
 import ChatMessage from './ChatMessage';
 import UserChatMessageEditor from './UserChatMessageEditor';
 import moduleStyles from './chatWorkspace.module.scss';
 import aichatI18n from '../locale';
-
+import {ChatCompletionMessage} from '../types';
+import {Role, Status} from '../constants';
 import {demoChatMessages} from './demoMessages'; // demo chat messages - remove when connected to backend
+import {openaiCompletion} from '../openai';
 
 const ChatWorkspace: React.FunctionComponent = () => {
+  const [storedMessages, setStoredMessages] =
+    useState<ChatCompletionMessage[]>(demoChatMessages);
+
+  const onSubmit = async (message: string) => {
```

Comment:
It would be good to break this function up a bit and move the non-React parts into a separate unit (or maybe even into the openai.ts file). One option could be to have a function like `async getChatCompletionMessage(systemPrompt: string, newMessage: string, chatMessages: ChatCompletionMessage[]): Promise<ChatCompletionResponse>` which can handle creating the list of messages and calling the api internally?

Reply:
Great idea! Implemented and just pushed. Things are kind of in an interim state since we have pending filters that will be implemented. Lmk what you think - thanks!